packages/notebook-extension/src/index.ts

```diff
@@ -948,6 +961,76 @@ function activateCodeConsole(
   });
 }
 
+/**
+ * Activate the output copying extension
+ */
+function activateCopyOutput(
+  app: JupyterFrontEnd,
+  translator: ITranslator,
+  tracker: INotebookTracker
+): void {
+  const trans = translator.load('jupyterlab');
+
+  /**
+   * Copy the contents of an HTMLElement to the system clipboard
+   */
+  function copyElement(e: HTMLElement): void {
+    const sel = window.getSelection();
+
+    if (sel == null) {
+      return;
+    }
+
+    // Save the current selection.
+    const savedRanges: Range[] = [];
+    for (let i = 0; i < sel.rangeCount; ++i) {
+      savedRanges[i] = sel.getRangeAt(i).cloneRange();
+    }
+
+    const range = document.createRange();
+    range.selectNodeContents(e);
+    sel.removeAllRanges();
+    sel.addRange(range);
+
+    document.execCommand('copy');
+
+    // Restore the saved selection.
+    sel.removeAllRanges();
+    savedRanges.forEach(r => sel.addRange(r));
+  }
+
+  app.commands.addCommand(CommandIDs.copyToClipboard, {
+    label: trans.__('Copy Output to Clipboard'),
+    execute: args => {
+      const cell = tracker.currentWidget?.content.activeCell as CodeCell;
+
+      if (cell == null) {
+        return;
+      }
+
+      const output = cell.outputArea.outputTracker.currentWidget;
+
+      if (output == null) {
+        return;
+      }
+
+      const outputAreaAreas = output.node.getElementsByClassName(
+        'jp-OutputArea-output'
+      );
+      if (outputAreaAreas.length > 0) {
+        const area = outputAreaAreas[0];
+        copyElement(area as HTMLElement);
+      }
+    }
+  });
+
+  app.contextMenu.addItem({
+    command: CommandIDs.copyToClipboard,
+    selector: '.jp-OutputArea-child',
+    rank: 0
+  });
```

Comment:
If there are two outputs only the first one will be copied because the implementation is cell-centric rather than on the level of outputs. An alternative implementation would use the event target (the actual output) and possibly have the plugin in `outputarea` package. This would also mean the class names could be used from constants where they are defined. I think that the most recent event target is stored in the application instance (not sure if it is public). It would be great for someone else to confirm if we should to go this route instead.